### DISCUSSION TITLE
Possibility of losing access of account with k-of-n auth from the loss of any public key

### DISCUSSION THREAD

**Discussion by yeptos** [aptos-labs/aptos-developer-discussions#388] [General]
I think one potential issue with the generalized k-of-n authentication scheme is that someone might lose access to their account even if they still have enough (≥ k) private keys to sign, but lose track of any of the public keys used to create the multi key. Concretely, one must know (i) all the public keys used to generate the auth key (ii) the order of the keys used (iii) the scheme of each key in order to sign a tx.

While this information is not secret, it can still be lost since it's not stored anywhere on the blockchain (unless a transaction has been signed at least once with each key) - only the auth key is registered during account creation or rotation. In practice, I can imagine scenarios where someone might mistakenly think, "I have these k private keys, so I can forget about the other n-k signers," only to later discover they've lost access because they can't sign without the public keys for the (n-k) signers.

I believe this could be viewed as a non-issue if we think it's the account owner's responsibility to keep record of all the associated public keys. Personally, I haven’t formed a strong opinion on this and would like to hear others’ thoughts.

**Comment by gregnazario** [comment]
For a MultiKey, the signature of the first transaction done by that account will have the public keys in the signature.  So as long as you submit one transaction after funding the account, you shouldn't lose the public keys

**Reply by yeptos** [reply]
In some situations it's already problematic that it requires the account to submit one transaction after rotating the key, since the public key information could be lost in between (we already discussed this irl but jotting down here for record)

**Reply by heliuchuan** [reply]
I agree that it would probably be good to allow for perhaps optionally allow for publishing of public key after rotation with a separate txn

**Reply by yeptos** [reply]
it would be nice if it can be optionally published with the key rotation transaction, not separately but atomically
